core/src/main/java/org/apache/brooklyn/core/mgmt/ha/OsgiManager.java

```diff
@@ -189,10 +193,11 @@ public static boolean isBundleNameEqualOrAbsent(CatalogBundle bundle, Bundle b)
                 Maybe<Bundle> bundle = findBundle(osgiBundle);
                 if (bundle.isPresent()) {
                     Bundle b = bundle.get();
-                    Class<T> clazz;
+                    Optional<String> strippedType = osgiClassPrefixer.stripMatchingPrefix(b, type);
+                    String typeToLoad = strippedType.isPresent() ? strippedType.get() : type;
```

Comment:
Good points @neykov. I'll strip the version from the prefix as well.

I'll let it match by just comparing the symbolic name (even if wrong version number) because that will better support upgrades/rollback of bundles, where the `symbolicName:version:className` has been persisted (I'd hope we miss out the version number in persisted state for that reason though).

I'd prefer not to try skipping the prefix "if the class comes from the bundles in the memento's catalog item". It makes sense at a high level, but I worry about situations where there are nested catalog items or sub-types (where different levels declare their libraries). We should handle such cases, but it feels to me like belt-and-braces if we include the bundle id here as well. If we do want to skip writing the prefix in such cases, let's look at that in a separate PR.